shims/spark310/src/main/scala/com/nvidia/spark/rapids/shims/spark310/ParquetCachedBatchSerializer.scala

```diff
@@ -369,31 +359,33 @@ class ParquetCachedBatchSerializer extends CachedBatchSerializer with Arm {
         // T01= {splitCol1(1), splitCol2(1),...,splitColn(1)}
         // ...
         // T0m= {splitCol1(m), splitCol2(m),...,splitColn(m)}
-        for (i <- splutVectors(0).indices) yield
-          new Table({
-            for (j <- splutVectors.indices) yield splutVectors(j)(i)
-          }: _*)
+        for (i <- splitVectors(0).indices)
+          withResource(new Table({
```

Comment:
There is no leak here, but it is rather confusing.  Could we make either creating the table or pulling the columns out for the table into a separate function so it is cleaner?

```
def makeTableForIndex(i: Int): Table = {
  val columns = splitVectors.indices.map(j => splitVectors(j)(i).toArray
  new Table(columns :_*)
}

for (i <- splitVectors(0).indices) {
  withResource(makeTableForIndex(i)) { table =>
    buffers += ParquetCachedBatch(writeTableToCachedBatch(table))
  }
}
```